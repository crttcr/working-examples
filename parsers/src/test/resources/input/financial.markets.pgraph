// Financial Markets provides a sample snapshot of what a model might look like
// for traded instruments.
//
//
## vertex.auto.id = GENERATE_SEQUENCE

(cs: ISSUE_TYPE { name = "Common Stock" })
(ps: ISSUE_TYPE { name = "Preferred Stock" })
(cp: ISSUE_TYPE { name = "Convertible Preferred Stock" })
(bd: ISSUE_TYPE { name = "Bond" })

(jpm: ISSUER
	{
	name = "JP Morgan Chase",
	label = "JPM"
	}
)

(jpm.cs: ISSUE { ticker = "JPM" })
(jpm.ps: ISSUE { ticker = "JPM.PS" })

(jpm.cs)--[:IS_A]->(cs)
(jpm.ps)--[:IS_A]->(ps)

(jpm)--[:ISSUED]->(jpm.cs)
(jpm)--[:ISSUED]->(jpm.ps)