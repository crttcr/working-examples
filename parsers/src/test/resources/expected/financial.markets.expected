## vertex.auto.id = GENERATE_SEQUENCE
VERTEX: cs
	LABEL: ISSUE_TYPE
		PROP name = "Common Stock"
VERTEX: ps
	LABEL: ISSUE_TYPE
		PROP name = "Preferred Stock"
VERTEX: cp
	LABEL: ISSUE_TYPE
		PROP name = "Convertible Preferred Stock"
VERTEX: bd
	LABEL: ISSUE_TYPE
		PROP name = "Bond"
VERTEX: jpm
	LABEL: ISSUER
		PROP name = "JP Morgan Chase"
		PROP label = "JPM"
VERTEX: jpm.cs
	LABEL: ISSUE
		PROP ticker = "JPM"
VERTEX: jpm.ps
	LABEL: ISSUE
		PROP ticker = "JPM.PS"
EDGE: IS_A
EDGE: IS_A
EDGE: ISSUED
EDGE: ISSUED
